File: android/app/src/main/kotlin/fr/batgard/thefocusapp/scenes/timer/businesslogic/Activity.kt
package fr.batgard.thefocusapp.scenes.timer.businesslogic

import kotlinx.serialization.Serializable

@Serializable
data class Activity(
        val type: ActivityType,
        val running: Boolean,
        val remainingTime: Duration
) {
    fun getNewToggledActivity() = Activity(type, !running, remainingTime) 
}

File: lib/scenes/timer/framework/Activity.dart
import 'dart:convert';

import 'package:json_annotation/json_annotation.dart';
import 'package:thefocusapp/scenes/timer/data/TimerDuration.dart';

part 'Activity.g.dart';

@JsonSerializable(nullable: false)
class CurrentActivity {
  String type;
  bool running;
  TimerDuration remainingTime;

  CurrentActivity(this.type, this.running, this.remainingTime);

  factory CurrentActivity.fromJson(Map<String, dynamic> json) => _$ActivityFromJson(json);

  Map<String, dynamic> toJson() => _$ActivityToJson(this);
}